bin/configs/go-defaults-for-required-vars.yaml

```diff
@@ -0,0 +1,6 @@
+generatorName: go
+outputDir: samples/client/others/go/defaults-for-required-vars
+inputSpec: modules/openapi-generator/src/test/resources/3_0/go/default-values-for-required-params.yaml
```

Comment:
isntead of using another test spec, what about adding the test cases to ` modules/openapi-generator/src/test/resources/3_0/go/petstore-with-fake-endpoints-models-for-testing-with-http-signature.yaml` instead? and update `./bin/configs/go-petstore.yaml` to enable `useDefaultValuesForRequiredVars` ?

Reply:
@wing328 I have made the changes. One test is failing, but I think it is unrelated to the changes here. Can you check once.

Reply:
Fixed